### DISCUSSION TITLE
Text that changes velocity

### DISCUSSION THREAD

**Discussion by fernandomartin777** [musescore/MuseScore#22458] [Ideas]
As I tested in nightly builds musescore 4.3 can now recognize velocity and play them correctly. However the only way to change velocity is selecting notes and going to their properties, which is very time consuming. Can you please, allow for some of the text categories to have a property to be used to change velocity?

**Comment by scorster** [comment]
Note velocity is essential in creating realistic and expressive playback. But in v4.2.1 Velocity is tucked away in General>Playback. Sure, it's just a single click deeper in the Properties panel. But with a note selected and the Playback area open, if you click *ANYTHING* other than a note you have to reopen the Playback section to edit a velocity value. 

Which raises a few questions: 

• Why does Properties>Playback have its own Notes section?
• Why does the Playback section have a disclosure triangle when *it houses only* a single Notes category? 
• Why not move Velocity into the Notes panel??? Like this:

        ![Velocity property in Notes panel](https://github.com/musescore/MuseScore/assets/121324720/9595fad1-32a2-4f49-bc08-e5fd714637b9)

**Comment by cbjeukendrup** [comment]
In 4.4 or 4.5, dynamics will affect the velocity instead of the volume, which should resolve this (to some extent).

**Reply by fernandomartin777** [reply]
It was that way in 3.6. I don't understand why they broke it. Furthermore, we could create custom dynamics according to our needs, like my personal dynamic palette for 3.x. It sounds much more realistic because I can lower or raise the dynamics slightly.
![20240419_231919](https://github.com/musescore/MuseScore/assets/116603050/9c0c99ba-aeb4-4d77-ae38-d48ab0c6583e)

Are you sure that they will bring it back in 4.4 or 4.5?

**Reply by cbjeukendrup** [reply]
> I don't understand why they broke it

They just didn't reimplement that yet in MuseScore 4's entirely new audio engine.

It is certain that "dynamics affecting velocity instead of volume" will come back: #19207 (currently scheduled as high priority for 4.4)

In principle, the velocity would be determined automatically based on which dynamic it is, but most likely the possibility to override it will come back too.

Also, in 4.5, an entire "automation" system will be introduced; this is a tool to specify velocity changes (and more) without having to add weird custom dynamics.

**Reply by fernandomartin777** [reply]
Hope they implement it as soon as possible.
